sources/packages/api/src/utilities/database.ts

```diff
@@ -18,3 +18,23 @@ export async function configureIdleTransactionSessionTimeout(
 ): Promise<void> {
   queryRunner.query(`SET idle_in_transaction_session_timeout = '${seconds}s'`);
 }
+
+/**
+ * This helper to to get the total count of the Raw for the pagination.
+ * For example, if there is a scenario to use typeorm `getRawMany`, typeorm
+ * doesn't have a utility to get the count of the raw entities (getManyAndCount
+ * will return count without innerJoins), For those cases we can use this helper
+ * function to get the actual total count of the raw data
+ * @param sqlQuery the sql query (result of typeorm .getSql()).
+ * if we want the total count, then pass the sql query without the
+ * pagination logic (i.e, without .skip(), .take(), .limit() or .offset())
+ * @param parameter pass the parameter of the sql query.
+ * @return the total count
+ */
+export async function getRawCount(
+  repo: Repository<any>,
+  sqlQuery: string,
+  parameter: any[],
+): Promise<[{ count: number }]> {
+  return repo.query(`SELECT COUNT(*) FROM (${sqlQuery}) AS count`, parameter);
```

Comment:
Instead of using the sql query, we can use .getRawMany() in the called query and can get the .length of it. So that we will have a single transaction.

Reply:
@guru-aot do you mean execute the _.getRawMany()_, retrieve all the records and then do the ".length"?
The parallel could happen either way, there is no need to change it.

Reply:
Sure, @andrewsignori-aot . As discussed, we can handle just the parallel call.